### PR TITLE
Address PR (#76) comments

### PR DESCRIPTION
**Pull request by viceroypenguin**

### CODE REVIEW

File: Source/SuperLinq/CopyTo.cs [hunk 1]
```diff
@@ -71,19 +78,30 @@ public static void CopyTo<TSource>(this IEnumerable<TSource> source, TSource[] a
...
-			&& n > array.Length)
+		if (source is ICollection<TSource> coll)
 		{
-			ThrowHelper.ThrowArgumentException(nameof(array), "Destination is not long enough.");
+			coll.CopyTo(array);
 		}
-
-		var i = 0;
-		foreach (var el in source)
+		else if (source is TSource[] arr)
```
Comment (Arithmomaniac) [inline Source/SuperLinq/CopyTo.cs:85]: `TSource[]` implements `ICollection<TSource>`. Remove this or (if it exists for performance) rearrange parameter order.

File: Source/SuperLinq/CopyTo.cs [hunk 2]
```diff
@@ -157,15 +175,31 @@ public static void CopyTo<TSource>(this IEnumerable<TSource> source, IList<TSour
 		}
 		else if (list is TSource[] array)
 		{
-			if (TryGetCollectionCount(source, out var n)
-				&& n > array.Length)
+			if (source is ICollection<TSource> coll)
```
Comment (Arithmomaniac) [inline Source/SuperLinq/CopyTo.cs:178]: This "ICollection but not Array" case is the only one that can't have the rest of the logic forwarded to the "Span target" version. After this block (or at least for the else) forward to the Span overload for clarity or create a private method to hold the common logic.

File: Source/SuperLinq/CopyTo.cs [hunk 3]
```diff
@@ -157,15 +175,31 @@ public static void CopyTo<TSource>(this IEnumerable<TSource> source, IList<TSour
 		}
 		else if (list is TSource[] array)
```
Comment (Arithmomaniac) [inline Source/SuperLinq/CopyTo.cs:176]: Forward this block to array overload (save code duplication)

**Review by Arithmomaniac — APPROVED** [review]

File: Source/SuperLinq/CopyTo.cs (hunk 2 above)
Comment (viceroypenguin) [reply Source/SuperLinq/CopyTo.cs]: private method makes sense. Done.

**Review by Arithmomaniac — APPROVED** [review]

**Pull request merged**